site/content/posts/2023-08-08-v0.25.0-impersonation-proxy-with-external-certs.md

```diff
@@ -0,0 +1,318 @@
+$ kind create cluster \
+    --config kind-expose-port.yaml \
+    --name kind-with-contour \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+```
+
+Install Contour (see https://projectcontour.io/getting-started/ for more details).
+
+```shell
+# From https://projectcontour.io/getting-started/
+$ kubectl apply \
+    --filename https://projectcontour.io/quickstart/contour.yaml \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+# Verify that the Contour pods are running
+$ kubectl get pods \
+    --namespace projectcontour \
+    --output wide \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+```
+
+Install Pinniped’s local-user-authenticator and add some sample users (see https://pinniped.dev/docs/tutorials/concierge-only-demo/ for more details).
+
+```shell
+# Install Pinniped's local-user-authenticator so that you have some demo users to play with
+$ kubectl apply \
+    --filename https://get.pinniped.dev/v0.25.0/install-local-user-authenticator.yaml \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+# Create a local user "pinny" with password "password123"
+$ kubectl create secret generic pinny \
+    --namespace local-user-authenticator \
+    --from-literal=groups=group-for-mtls \
+    --from-literal=passwordHash=$(htpasswd -nbBC 10 x password123 | sed -e "s/^x://") \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+# The local-user-authenticator will serve a TLS endpoint for the Concierge to talk to,
+# so you need to configure the Concierge's webhook to verify TLS with the appropriate CA.
+# Just make sure this next command does print out the TLS secret, which can take a few seconds to generate.
+$ kubectl get secret local-user-authenticator-tls-serving-certificate \
+    --namespace local-user-authenticator \
+    --output jsonpath={.data.caCertificate} \
+    --kubeconfig kind-with-contour.kubeconfig.yaml \
+    | tee local-user-authenticator-ca.pem.b64
+```
+
+Install `cert-manager`:
+
+```shell
+$ kubectl apply \
+    --filename https://github.com/cert-manager/cert-manager/releases/download/v1.12.3/cert-manager.yaml \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+```
+
+Configure a `cert-manager` certificate for the impersonation proxy to serve TLS:
+
+```shell
+$ cat << EOF > self-signed-cert.yaml
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: cert-manager
+---
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: selfsigned-issuer
+spec:
+  selfSigned: {}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: my-selfsigned-ca
+  namespace: cert-manager
+spec:
+  isCA: true
+  commonName: my-selfsigned-ca
+  secretName: impersonation-proxy-self-signed-ca
+  privateKey:
+    algorithm: ECDSA
+    size: 256
+  issuerRef:
+    name: selfsigned-issuer
+    kind: ClusterIssuer
+    group: cert-manager.io
+---
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: my-ca-issuer
+spec:
+  ca:
+    secretName: root-secret
```

Comment:
This should probably be `impersonation-proxy-self-signed-ca`

Reply:
Yup to match above 👍

Reply:
done, see next push